tests/test_fock.py

```diff
@@ -625,6 +628,185 @@ def circuit(n, a):
+            qml.Displacement(a, phi, wires=0)
+            qml.Displacement(a, phi, wires=1)
+            return qml.probs(wires=[0, 1])
+
+        a = 0.4
+        phi = -0.12
+
+        c = np.arange(cutoff)
+        d = np.arange(cutoff)
+        n0, n1 = np.meshgrid(c, d)
+        n0 = n0.flatten()
+        n1 = n1.flatten()
+
+        # differentiate with respect to parameter a
+        res_F = circuit.jacobian([a, phi], wrt={0}, method="F").flat
+        expected_gradient = 2 * (a **(-1 + 2*n0 + 2*n1)) * np.exp(-2*a ** 2) * (-2*a ** 2 + n0 + n1) / (fac(n0) * fac(n1))
+        assert np.allclose(res_F, expected_gradient, atol=tol, rtol=0)
+
+        # differentiate with respect to parameter phi
+        res_F = circuit.jacobian([a, phi], wrt={1}, method="F").flat
+        expected_gradient = 0
+        assert np.allclose(res_F, expected_gradient, atol=tol, rtol=0)
+
+    def test_finite_diff_squeezed(self, tol):
+        """Test that the jacobian of the probability for a squeezed state is
+        approximated well with finite differences"""
+        cutoff = 10
+
+        dev = qml.device("strawberryfields.fock", wires=1, cutoff_dim=cutoff)
+
+        @qml.qnode(dev)
+        def circuit(a, phi):
+            qml.Squeezing(a, phi, wires=0)
+            return qml.probs(wires=[0])
+
+        r = 0.4
+        phi = -0.12
+
+        n = np.arange(cutoff, dtype=float)
+
+        hermite_coeffs = np.array([1, 0, -2, 0, 12, 0, -120, 0, 1680, 0])
+
+        def sech(z):
+            return 1 / np.cosh(z)
+
+        def csch(z):
+            return 1 / np.sinh(z)
+
+        expected_gradient = hermite_coeffs **2 *(2**(-1 - n)*np.abs(np.tanh(r))**n*(1 + 2*n - np.cosh(2*r))*csch(r)*sech(r)**2)/fac(n)
+
+        # differentiate with respect to parameter r
+        res_F = circuit.jacobian([r, phi], wrt={0}, method="F").flat
+        assert np.allclose(res_F, expected_gradient, atol=tol, rtol=0)
+
+        # differentiate with respect to parameter phi
+        res_F = circuit.jacobian([r, phi], wrt={1}, method="F").flat
+        #expected_gradient = hermite_coeffs **2 * 1j*2**(-n)*np.abs(np.tanh(r))**n * sech(r) / gamma(n)
+        #print(circuit(r, phi), circuit.jacobian([r, phi], wrt={1}, method="F"), expected_gradient)
+        #assert np.allclose(res_F, expected_gradient, atol=tol, rtol=0)
+
+    def test_analytic_diff_error(self, tol):
+        """Test that the analytic gradients are not supported when returning
+        Fock state probabilities."""
+        cutoff = 10
+
+        dev = qml.device("strawberryfields.fock", wires=1, cutoff_dim=cutoff)
+
+        @qml.qnode(dev)
+        def circuit(a, phi):
+            qml.Displacement(a, phi, wires=0)
+            return qml.probs(wires=[0])
+
+        a = 0.4
+        phi = -0.12
+        with pytest.raises(ValueError, match="Analytic gradients are not supported for circuits extracting Fock state probabilities"):
+            res_F = circuit.jacobian([a, phi], wrt={0}, method="A").flat
```

Comment:
This was added as per the error featured in https://github.com/XanaduAI/pennylane/pull/699